src/test/java/io/cdap/plugin/sfmc/sink/DataExtensionOutputFormatTest.java

```diff
@@ -0,0 +1,91 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DataExtensionClient.class, ETClient.class})
+public class DataExtensionOutputFormatTest {
+  public static final String CLIENT_ID = "cdap.sfmc.client.id";
+  public static final String CLIENT_SECRET = "cdap.sfmc.client.secret";
+  public static final String AUTH_ENDPOINT = "cdap.sfmc.auth.endpoint";
+  public static final String SOAP_ENDPOINT = "cdap.sfmc.soap.endpoint";
+  public static final String DATA_EXTENSION_KEY = "cdap.sfmc.data.extension.key";
+  public static final String MAX_BATCH_SIZE = "cdap.sfmc.max.batch.size";
+  public static final String FAIL_ON_ERROR = "cdap.sfmc.fail.on.error";
+  public static final String OPERATION = "cdap.sfmc.operation";
+  public static final String TRUNCATE = "cdap.sfmc.truncate";
+
+  @Test
+  public void testCheckOutSpecs() {
+    JobContext context1 = Mockito.mock(JobContext.class);
+    DataExtensionOutputFormat outputFormat = Mockito.mock(DataExtensionOutputFormat.class);
+    Mockito.verify(outputFormat, Mockito.times(0)).checkOutputSpecs(context1);
+  }
+
+  @Test
+  public void testNeedsTaskCommit() throws IOException {
+    TaskAttemptContext taskAttemptContext = Mockito.mock(TaskAttemptContext.class);
+    DataExtensionOutputFormat outputFormat = new DataExtensionOutputFormat();
+    Assert.assertFalse(outputFormat.getOutputCommitter(taskAttemptContext).needsTaskCommit(taskAttemptContext));
+  }
+
+  @Test
+  public void testGetRecordWriter() throws Exception {
+    TaskAttemptContext context = Mockito.mock(TaskAttemptContext.class);
+    Configuration conf = new Configuration();
+    conf.set(CLIENT_ID, "clientid");
+    conf.set(CLIENT_SECRET, "clientSecret");
+    conf.set(AUTH_ENDPOINT, "authEndPoint");
+    conf.set(SOAP_ENDPOINT, "soapEndPoint");
+    conf.set(OPERATION, "INSERT");
+    conf.set(DATA_EXTENSION_KEY, "DE");
+    conf.set(MAX_BATCH_SIZE, "500");
+    conf.set(FAIL_ON_ERROR, "true");
+    conf.set(TRUNCATE, "true");
+    String dataExtensionKey = "DE";
+    Mockito.when(context.getConfiguration()).thenReturn(conf);
+    ETClient etClient = PowerMockito.mock(ETClient.class);
+    PowerMockito.whenNew(ETClient.class).withArguments(Mockito.anyString()).thenReturn(etClient);
+    PowerMockito.mockStatic(DataExtensionClient.class);
+    DataExtensionClient client = PowerMockito.mock(DataExtensionClient.class);
+    PowerMockito.whenNew(DataExtensionClient.class).withArguments(Mockito.any(), Mockito.anyString()).thenReturn(client);
+    PowerMockito.when(DataExtensionClient.create(dataExtensionKey, "clientid", "clientSecret", "authEndPoint",
+                                                 "soapEndPoint")).thenReturn(client);
+    List<ETDataExtensionColumn> list = new ArrayList<>();
+    ETDataExtensionColumn column = new ETDataExtensionColumn();
+    column.setName("name");
+    list.add(column);
+    DataExtensionInfo info = new DataExtensionInfo("key", list);
```

Comment:
this seems to be unused? what is this test doing?

Reply:
This test is useful, as it is testing that whether the recordWriter is getting created or not by passing the right values, inclusive of getOrError() method of same class.